x-pack/plugins/osquery/server/lib/fleet_integration.ts

```diff
@@ -5,13 +5,65 @@
  * 2.0.
  */
 
-import type { SavedObjectReference, SavedObjectsClient } from '@kbn/core/server';
-import { filter, map } from 'lodash';
+import type {
+  ElasticsearchClient,
+  SavedObjectReference,
+  SavedObjectsClient,
+} from '@kbn/core/server';
+import { filter, map, orderBy } from 'lodash';
+import deepequal from 'fast-deep-equal';
+import { satisfies } from 'semver';
 import type { PostPackagePolicyPostDeleteCallback } from '@kbn/fleet-plugin/server';
+import type { UpdatePackagePolicy } from '@kbn/fleet-plugin/common';
 import { AGENT_POLICY_SAVED_OBJECT_TYPE } from '@kbn/fleet-plugin/common';
 import { packSavedObjectType } from '../../common/types';
 import { OSQUERY_INTEGRATION_NAME } from '../../common';
 
+export const getPackagePolicyUpdateCallback =
+  (esClient: ElasticsearchClient) => async (updatePackagePolicy: UpdatePackagePolicy) => {
+    if (
+      updatePackagePolicy.package?.name === OSQUERY_INTEGRATION_NAME &&
+      satisfies(updatePackagePolicy.package?.version ?? '', '>=1.6.0')
+    ) {
+      const mapping = await esClient.indices.getMapping({
+        index: `logs-${OSQUERY_INTEGRATION_NAME}.result-*`,
+      });
+      // Sort by index name to get the latest index
+      const dataStreamMapping = orderBy(Object.entries(mapping), [0], 'desc')?.[0][1]?.mappings
+        ?.properties?.data_stream;
+
+      if (
+        dataStreamMapping &&
+        deepequal(dataStreamMapping, {
+          properties: {
+            dataset: {
+              type: 'constant_keyword',
+              value: 'generic',
+            },
+            namespace: {
+              type: 'constant_keyword',
+              value: 'default',
+            },
+            type: {
+              type: 'constant_keyword',
+              value: 'osquery',
+            },
+          },
+        })
+      ) {
+        try {
+          esClient.indices.rollover({
+            alias: `logs-${OSQUERY_INTEGRATION_NAME}.result-default`,
```

Comment:
this will solve the rollover issue for users who are only using the `default` namespace.  We would need to expand this to also rollover custom namespaces as specified in the different Agent policies.